[Elastic Agent] Panic error while enrolling an agent with a revoked token

## Runtime metadata
- Version: Latest 8.0.0-SNAPSHOT
- Operating System: Centos 7 Docker

## Steps to Reproduce: 
1. setup Fleet
1. Create a token
1. Start a Centos 7 docker container in the same network as Kibana, ES and package-registry
1. Install latests SNAPSHOT for the agent in the container (https://www.elastic.co/guide/en/ingest-management/current/elastic-agent-installation.html)
1. Use the already created token to enroll the agent, using the insecure flag, as Kibana runs on HTTP
1. Revoke the token in Kibana
1. Destroy the centos7 service
1. Start a Centos 7 docker container in the same network as Kibana, ES and package-registry
1. Install latests SNAPSHOT for the agent in the container (https://www.elastic.co/guide/en/ingest-management/current/elastic-agent-installation.html)
1. Use the already created token to enroll the agent, using the insecure flag, as Kibana runs on HTTP

#### Expected behavior
The agent is not enrolled with no panic errors on agent side

#### Actual behavior
The agent is not enrolled but panic errors are thrown on agent side. See the following stacktrace:

```stacktrace
The Elastic Agent is currently in BETA and should not be used in production
2020-07-16T06:33:30.445Z        DEBUG   kibana/client.go:170    Request method: POST, path: /api/ingest_manager/fleet/agents/enroll
panic: runtime error: comparing uncomparable type errors.agentError

goroutine 1 [running]:
github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/cmd.enroll(0xc00031ade0, 0xc000338c80, 0xc00031ae10, 0xc000275180, 0x2, 0x4, 0xcaff9d, 0x4)
        /go/src/github.com/elastic/beats/x-pack/elastic-agent/pkg/agent/cmd/enroll.go:125 +0xac7
github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/cmd.newEnrollCommandWithArgs.func1(0xc000338c80, 0xc000275180, 0x2, 0x4)
        /go/src/github.com/elastic/beats/x-pack/elastic-agent/pkg/agent/cmd/enroll.go:36 +0x6d
github.com/spf13/cobra.(*Command).execute(0xc000338c80, 0xc0002750c0, 0x4, 0x4, 0xc000338c80, 0xc0002750c0)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc000338500, 0x6, 0x6, 0xc00031ade0)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ea
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
main.main()
        /go/src/github.com/elastic/beats/x-pack/elastic-agent/main.go:20 +0x138
```

## Related issues
- Discovered while testing https://github.com/elastic/e2e-testing/pull/149, more specifically this use case https://github.com/elastic/e2e-testing/blob/master/e2e/_suites/ingest-manager/features/fleet_mode_agent.feature#L43-L47

--- a/x-pack/elastic-agent/CHANGELOG.asciidoc
+++ b/x-pack/elastic-agent/CHANGELOG.asciidoc
@@ -50,6 +50,7 @@
 - Properly stops subprocess on shutdown {pull}19567[19567]
 - Forward revision number of the configuration to the endpoint. {pull}19759[19759]
 - Remove support for logs type and use logfile {pull}19761[19761]
+- Avoid comparing uncomparable types on enroll {issue}19976[19976]
 
 ==== New features
 

--- a/x-pack/elastic-agent/pkg/agent/cmd/enroll.go
+++ b/x-pack/elastic-agent/pkg/agent/cmd/enroll.go
@@ -122,7 +122,7 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args
 
 	backExp := backoff.NewExpBackoff(signal, 60*time.Second, 10*time.Minute)
 
-	for err == fleetapi.ErrTooManyRequests {
+	for errors.Is(err, fleetapi.ErrTooManyRequests) {
 		fmt.Fprintln(streams.Out, "Too many requests on the remote server, will retry in a moment.")
 		backExp.Wait()
 		fmt.Fprintln(streams.Out, "Retrying to enroll...")

